crates/solver-core/src/context.rs

```
@@ -0,0 +1,428 @@
+            }
+        };
+
+        // 2. Fetch chain data for each relevant chain
+        let mut chain_data = HashMap::new();
+        for chain_id in &involved_chains {
+            if let Ok(data) = self.delivery.get_chain_data(*chain_id).await {
+                chain_data.insert(*chain_id, data);
+            } else {
+                tracing::warn!(
+                    chain_id = chain_id,
+                    intent_id = %intent.id,
+                    "Failed to fetch chain data, decision may be suboptimal"
+                );
+            }
+        }
+
+        // 3. Get solver balances for relevant chains/tokens
+        let solver_balances = self.fetch_solver_balances(&involved_chains).await?;
+
+        tracing::info!("Built execution context");
+        tracing::info!("Chain data: {:?}", chain_data);
+        tracing::info!("Solver balances: {:?}", solver_balances);
+
+        Ok(ExecutionContext {
+            chain_data,
+            solver_balances,
+            timestamp,
+        })
+    }
+
+    /// Extracts chain IDs involved in the intent based on its standard.
+    ///
+    /// Parses the intent data to determine which chains are involved
+    /// in the cross-chain operation.
+    fn extract_chains_from_intent(&self, intent: &Intent) -> Result<Vec<u64>, SolverError> {
+        tracing::debug!(
+            intent_id = %intent.id,
+            standard = %intent.standard,
+            "Attempting to extract chains from intent"
+        );
+
+        let result = match intent.standard.as_str() {
+            "eip7683" => self.extract_eip7683_chains(&intent.data),
+            _ => {
+                tracing::warn!(
+                    standard = %intent.standard,
+                    intent_id = %intent.id,
+                    "Unsupported intent standard, using fallback chain detection"
+                );
+                // Fallback: try to extract from common fields
+                self.extract_fallback_chains(&intent.data)
```

I think for now we should just throw an unsupported standard error instead of attempting some custom fallback logic.